src/libcore/num/mod.rs

```diff
@@ -1321,7 +1321,11 @@ macro_rules! int_impl {
         #[stable(feature = "rust1", since = "1.0.0")]
         #[inline]
         pub fn abs(self) -> $T {
-            if self.is_negative() { -self } else { self }
+            if self.is_negative() {
+                self.wrapping_neg()
```

Comment:
the docs that lie *just* out of reach of the diff say:
> `Int::min_value()` will be returned if the number is `Int::min_value()`.

So I was just preserving that.  (Plus I think there are tests that check it.)

But yeah, seems like a Q for https://github.com/rust-lang/rfcs/pull/1017